scrapy/exporters.py

```diff
@@ -238,9 +244,14 @@ def export_item(self, item):
             self._headers_not_written = False
             self._write_headers_and_set_fields_to_export(item)
 
-        fields = self._get_serialized_fields(item, default_value='',
-                                             include_empty=True)
-        values = list(self._build_row(x for _, x in fields))
+        serialized_fields = self._get_serialized_fields(item, default_value='', include_empty=True)
+        fields = self._get_fields_from_item(item)
+        if not self._data_loss_already_warned:
+            if self.fields_to_export is None:
+                logger.warning("Data loss detected when exporting items -- This message won't be shown for further items")
+            elif set(fields) != set(self.fields_to_export):
+                logger.info("Data loss detected when exporting items -- This message won't be shown for further items")
```

Comment:
Why is `self._data_loss_already_warned` not set to `True` here? Won’t that cause this message to happen on every exported item?

Reply:
Yes, I don't know how I missed that. I apologize.